spring-integration-core/src/main/java/org/springframework/integration/store/SimpleMessageStore.java

```diff
@@ -208,6 +239,12 @@ public MessageGroup addMessageToGroup(Object groupId, Message<?> message) {
 				if (group == null) {
 					group = new SimpleMessageGroup(groupId);
 					this.groupIdToMessageGroup.putIfAbsent(groupId, group);
+					this.groupToUpperBound.putIfAbsent(groupId, new UpperBound(this.groupCapacity));
+				}
+				if (!this.groupToUpperBound.get(groupId).tryAcquire(this.upperBoundTimeout)) {
```

Comment:
@garyrussell , unfortunately  we don't have here `InterruptedException`.
See `UpperBound.tryAcquire()` source code:
```
catch (InterruptedException e) {
	Thread.currentThread().interrupt();
	return false;
}
```

Or am I missing anything here?